pkg/types/var.go

```diff
@@ -71,6 +72,19 @@ func (v *Var) defaulting() {
 	}
 }
 
+// VarEquals returns true if var a and b are Equals.
+func (v Var) DeepEqual(other Var) bool {
+	v.ObjRef.GVK()
+	if v.FieldRef.FieldPath == "" {
```

Comment:
I don't think you need to do this check here since the defaulting is already done in `Absorb` before calling this function.

Reply:
@Liujingfang1 Changed the code to Defaulting() on both members before invoking reflect.DeepEquals